src/entrypoints/recorder.ts

```diff
@@ -420,13 +422,18 @@ function _tryReadBody(r: Request | Response): Promise<string> {
     // eslint-disable-next-line compat/compat
     return new Promise((resolve, reject) => {
         const timeout = setTimeout(() => resolve('[SessionReplay] Timeout while trying to read body'), 500)
-        r.clone()
-            .text()
-            .then(
-                (txt) => resolve(txt),
-                (reason) => reject(reason)
-            )
-            .finally(() => clearTimeout(timeout))
+        try {
+            r.clone()
+                .text()
+                .then(
+                    (txt) => resolve(txt),
+                    (reason) => reject(reason)
+                )
+                .finally(() => clearTimeout(timeout))
+        } catch (e) {
```

Comment:
this is a genuine fix for an actual bug, not just a test addition

errors in `.text()` are caught in the `then` and rejected

but `r.clone()` can throw and we weren't catching that